doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    \subsection{Multimodal log management}
+    In robotics, it is common to use the word \q{logging} to refer to the process of
+    writing multimodal diagnostic messages to network or disk for followup
+    analysis. Perhaps by intent, the term serves to emphasize the parallels with
+    traditional server application logging, i.e applications log text, robots log
+    point clouds.
+
+    Server application log management platforms are plentiful and follow
+    well-established patterns. Some examples of these include Sumo logic
+    \cite{sumologic}, Datadog logging \cite{datadoglog}, and the log management
+    interfaces of the three major cloud providers: Google Cloud Logging \cite{gcplog}, AWS
+    Cloudwatch Logs \cite{cloudwatch}, and Azure Monitor Logs \cite{azlog}.
+
+    From the user’s point of view these log management solutions are mostly
+    interchangeable. The core interface consists of, (1) Search controls:
+    dropdowns for error levels, date picking, application names, tags, and a text
+    box for \q{advanced} searches in a limited query language. (2) A listing of
+    log lines matching the search results in descending timestamp order. (3) A
+    \q{view in context} function, usually accessed by right-clicking a
+    particular log line in the listing. This function will show the log line of
+    interest highlighted (or \q{pinned}) in a wider scope, perhaps by dropping the
+    search restriction. For instance if the original query was restricted to a
+    single application, the \q{context} may now include all applications. This
+    function is used for tracing cascading failures and events across services.
+
+    Users of multimodal logs have a similar set of requirements, with additional
+    challenges.
+
+    First, there are more kinds of content-based search, such as geographical
+    search, image similarity search, or other kinds of search unanticipated by
+    the database designer (i.e UDFs), and perhaps a more frequent need to
+    correlate multiple signals in a single search. The need for timestamp-based
+    search controls and filtering on high-level dimensions remains as great as
+    before.
+
+    Second, the meaning of \q{listing log lines} or \q{filling a screen with
+    logs} becomes a lot less clear, and is an interesting question in user
+    interface design. Log lines can still be listed as JSON, with bytes-valued
+    fields rendered in base64 or similar – but more creative approaches can
+    likely do better by incorporating images, pointcloud visualizations, maps,
+    audio, and more.
+
+
+    Third, text rendering is clearly insufficient for the \q{view in context}
+    function, which now represents replaying \q{what the robot saw} at the
+    associated time. Instead of a pinned message in a larger resultset, \q{view
+    in context} is implemented using a complex visualization application such as
+    Foxglove.
+
+    Beyond these parallels, there are other points of divergence:
+
```

Comment:
I think we could benefit from having the motivation presented more clearly? This could probably be achieved with some rework / moving around

Reply:
TL;DR; could we have the below text be more visible?